climakitae/explore/amy.py

```diff
@@ -129,36 +133,125 @@ def retrieve_meteo_yr_data(
+    Tuple[xr.Dataset, xr.Dataset]
+        (historic_data, future_data) - Historical data at 1.2°C warming,
+        and future data at specified warming levels.
 
-    # Convert Julian date index to Month-Day format
-    # Use 2024 as year if we have 366 days (leap year), otherwise use 2023
-    year = 2024 if len(df) == 366 else 2023
-    new_index = [
-        julianDay_to_date(julday, year=year, str_format="%b-%d") for julday in df.index
-    ]
-    df.index = pd.Index(new_index, name="Day of Year")
-    return df
+    Raises
+    ------
+    ValueError
+        If invalid parameter keys are provided.
+
+    Example
+    -------
+    >>> historic_data, future_data = retrieve_profile_data(
+    ...     variable="Air Temperature at 2m",
+    ...     resolution="45 km",
+    ...     scenario=["SSP 3-7.0"],
+    ...     warming_level=[1.5, 2.0, 3.0],
+    ...     units="degF"
+    ... )
+
+    >>> historic_data, future_data = retrieve_profile_data(
+    ...     warming_level=[2.0]
+    ... )
+
+    Notes
+    -----
+    Historical data is always retrieved for warming level = 1.2°C.
+    Future data uses user-specified warming levels or defaults.
+    """
+    no_delta = kwargs.pop("no_delta", False)
+    # Define allowed inputs with types and defaults
+    ALLOWED_INPUTS = {
+        "variable": (str, "Air Temperature at 2m"),
+        "resolution": (str, "3 km"),
+        "warming_level": (list, [1.2]),
+        "cached_area": ((str, list), None),
+        "units": (str, "degF"),
+        "latitude": ((float, tuple), None),
+        "longitude": ((float, tuple), None),
+    }
+
+    # if the user does not enter warming level the analysis is a moot point
+    # because the historical data is always at 1.2C
+    REQUIRED_INPUTS = ["warming_level"]
+    for req in REQUIRED_INPUTS:
+        if req not in kwargs:
+            raise ValueError(f"Missing required input: '{req}'")
+
+    # Validate input keys
+    invalid_keys = set(kwargs.keys()) - set(ALLOWED_INPUTS.keys())
+    if invalid_keys:
+        raise ValueError(
+            f"Invalid input(s): {list(invalid_keys)}. "
+            f"Allowed inputs are: {list(ALLOWED_INPUTS.keys())}"
+        )
+
+    # Validate input types
+    for key, value in kwargs.items():
+        expected_type, _ = ALLOWED_INPUTS[key]
+        if not isinstance(value, expected_type):
+            raise TypeError(
+                f"Parameter '{key}' must be of type {expected_type.__name__}, "
+                f"got {type(value).__name__}"
+            )
+
+    # Set default parameters for data retrieval
+    get_data_params = {
+        "variable": kwargs.get("variable", "Air Temperature at 2m"),
+        "resolution": kwargs.get("resolution", "3 km"),
+        "downscaling_method": "Dynamical",  # must be WRF, cannot be LOCA
+        "timescale": kwargs.get(
+            "timescale", "hourly"
+        ),  # must be hourly for 8760 analysis
+        "area_average": "Yes",
+        "units": kwargs.get("units", None),
```

Comment:
```suggestion
        "units": kwargs.get("units", "degF"),
```
for default degF units